components/FieldTripListPage/index.js

```diff
@@ -13,34 +13,44 @@ import {
   LinkButton,
   RemoveButton,
   CustomLink,
+  EditButton,
 } from "../component.styled";
 import { fieldtrips as mockFieldtrips } from "../../db/mocked/index.js";
 import useLocalStorageState from "use-local-storage-state";
 
 const FieldTripListPage = () => {
   const router = useRouter();
-  const [showPopup, setShowPopup] = useState(false);
+  const [showAddPopup, setShowAddPopup] = useState(false);
+  const [showEditPopup, setShowEditPopup] = useState(false);
   const [newFieldTripName, setNewFieldTripName] = useState("");
   const [newFieldTripDate, setNewFieldTripDate] = useState("");
   const [fieldtrips, setFieldtrips] = useLocalStorageState("fieldTrips", {
     defaultValue: mockFieldtrips,
   });
 
+  const [editingFieldTripId, setEditingFieldTripId] = useState(null);
+  const [editedFieldTripName, setEditedFieldTripName] = useState("");
+  const [editedFieldTripDate, setEditedFieldTripDate] = useState("");
+
+  const handleFieldTripLinkClick = (fieldTripId) => {
+    localStorage.setItem("currentFieldTripId", fieldTripId); // set currentFieldTripId in local storage for navigation
+  };
+
   const handleAddFieldTrip = () => {
+    //handle add field trip which happens when user clicks on add button
     if (newFieldTripName.trim() !== "" && newFieldTripDate.trim() !== "") {
-      // if input is not empty
-      const newDate = formatDate(newFieldTripDate); // format new date
+      const newDate = formatDate(newFieldTripDate);
 
       const newFieldTrip = {
-        id: fieldtrips.length + 1, // add new fieldtrip id
-        fieldtripname: newFieldTripName, // add new fieldtrip name
-        fieldtripdate: newDate, // add new fieldtrip date
-        outcrops: [], // add new fieldtrip outcrops
+        id: fieldtrips.length + 1, // added fieldtrip id is dependant on the length of fieldtrips
```

Comment:
I sugest using uuid or any otehr id-generating library. 
imagine you are deleting one item later - its very likely that there are two same ids very fast.

Reply:
Yeah, thanks kinda huge problem after testing it a bit solved it without a library or atleast i hope so.